manifests/service.pp

```diff
@@ -21,6 +21,17 @@
   $install_dir       = $wso2base::install_dir
   $carbon_home       = $wso2base::carbon_home
   $autostart_service = $wso2base::autostart_service
+  $file_list_copy_without_refresh = $wso2base::file_list_copy_without_refresh
+
+  if ($file_list_copy_without_refresh != undef and size($file_list_copy_without_refresh) > 0) {
+    wso2base::push_files {
+      $file_list_copy_without_refresh:
+        owner       => $wso2_user,
+        group       => $wso2_group,
+        carbon_home => $carbon_home,
+        wso2_module => $caller_module_name
```

Comment:
I'm more concerned with the effect of keeping this without binding to any other resource that would result in this resource being executed parallelly to some other resource. It would be better to be explicit in the required order.